editor/src/components/canvas/canvas-selection.spec.browser2.tsx

```diff
@@ -0,0 +1,310 @@
+        }}
+        data-label='Blue'
+      />
+    </div>
+  );
+};
+
+export var storyboard = (
+  <Storyboard data-uid="storyboard">
+    <Scene
+      data-uid="scene-1"
+      style={{ position: "absolute", left: 0, top: 0, width: 375, height: 812 }}
+    >
+      <App data-uid="app" />
+    </Scene>
+  </Storyboard>
+);
+`
+}
+
+function createAndRenderModifiedProject(modifiedFiles: {
+  [filename: string]: string
+}): Promise<EditorRenderResult> {
+  const project = createModifiedProject(modifiedFiles)
+  return renderTestEditorWithModel(project, 'await-first-dom-report')
+}
+
+function createExampleProject(
+  positionSetting: string | null,
+  elementTopShifts: [number, number, number],
+): Promise<EditorRenderResult> {
+  return createAndRenderModifiedProject({
+    [StoryboardFilePath]: exampleProject(positionSetting, elementTopShifts),
+  })
+}
+
+async function checkOverlappingElements(
+  positionSetting: string | null,
+  elementTopShifts: [number, number, number],
+  focusedOn: ElementPath | null,
+  targetDivId: string,
+  cmdPressed: boolean,
+  x: number,
+  y: number,
+  expectedSelectedElements: Array<ElementPath>,
+): Promise<void> {
+  const renderResult = await createExampleProject(positionSetting, elementTopShifts)
+
+  const targetDiv = renderResult.renderedDOM.getByTestId(targetDivId)
+  const targetDivBounds = targetDiv.getBoundingClientRect()
+
+  const canvasControlsLayer = renderResult.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+  if (focusedOn != null) {
+    await renderResult.dispatch([setFocusedElement(focusedOn)], true)
+  }
+
+  act(() => {
+    fireEvent(
+      canvasControlsLayer,
+      new MouseEvent('mousedown', {
+        detail: 1,
+        bubbles: true,
+        cancelable: true,
+        metaKey: cmdPressed,
+        clientX: targetDivBounds.left + x,
+        clientY: targetDivBounds.top + y,
+        buttons: 1,
+      }),
+    )
+  })
+
+  const newlySelectedElements = renderResult.getEditorState().editor.selectedViews
+  expect(newlySelectedElements).toEqual(expectedSelectedElements)
+}
+
+const redDivElementPath = EP.elementPath([
+  ['storyboard', 'scene-1', 'app'],
+  ['other-app-root', 'div-red'],
+])
+
+const greenDivElementPath = EP.elementPath([
+  ['storyboard', 'scene-1', 'app'],
+  ['other-app-root', 'div-green'],
+])
+
+const blueDivElementPath = EP.elementPath([
+  ['storyboard', 'scene-1', 'app'],
+  ['other-app-root', 'div-blue'],
+])
+
+const scene1ElementPath = EP.elementPath([['storyboard', 'scene-1']])
+
+const otherAppRootElementPath = EP.elementPath([
+  ['storyboard', 'scene-1', 'app'],
+  ['other-app-root'],
+])
+
+const zeroElementTopShifts: [number, number, number] = [0, 0, 0]
+
+const relativeElementTopShifts: [number, number, number] = [0, -100, -200]
+
+const overlappingElementTestCases: Array<
+  [
+    string | null,
+    [number, number, number],
+    ElementPath | null,
+    string,
+    boolean,
+    number,
+    number,
+    Array<ElementPath>,
+  ]
```

Comment:
Yeah, this started out like this because of `it.each`, but we can't use that because reasons.

Reply:
Yeah because we use Mocha with Karma (which doesn't have `it.each`), but this is what I've done elsewhere with a Karma test: https://github.com/concrete-utopia/utopia/blob/master/editor/src/components/inspector/common/inspector-end-to-end-tests.spec.browser2.tsx#L2077